src/Str.php

```diff
@@ -197,6 +197,29 @@ public static function isValidPhpVariableName($name)
         return (bool) preg_match('/^[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*$/', $name, $matches);
     }
 
+    public static function isValidPhpClassName(string $className): bool
+    {
+        $reservedKeywords = ['__halt_compiler', 'abstract', 'and', 'array',
+            'as', 'break', 'callable', 'case', 'catch', 'class',
+            'clone', 'const', 'continue', 'declare', 'default', 'die', 'do',
+            'echo', 'else', 'elseif', 'empty', 'enddeclare', 'endfor',
+            'endforeach', 'endif', 'endswitch', 'endwhile', 'eval',
+            'exit', 'extends', 'final', 'for', 'foreach', 'function',
+            'global', 'goto', 'if', 'implements', 'include',
+            'include_once', 'instanceof', 'insteadof', 'interface', 'isset',
+            'list', 'namespace', 'new', 'or', 'print', 'private',
+            'protected', 'public', 'require', 'require_once', 'return',
+            'static', 'switch', 'throw', 'trait', 'try', 'unset',
+            'use', 'var', 'while', 'xor', '__CLASS__', '__DIR__', '__FILE__',
+            '__FUNCTION__', '__LINE__', '__METHOD__', '__NAMESPACE__', '__TRAIT__',
+            'int', 'float', 'bool', 'string', 'true', 'false', 'null', 'void',
+            'iterable', 'object',
```

Comment:
> 'int', 'float', 'bool', 'string', 'true', 'false', 'null', 'void',
            'iterable', 'object',

This words are allowed to use in Namespace, anyway it will be very strange to use it :-)